lib/Backend/IR.cpp

```diff
@@ -2880,107 +2880,134 @@ IR::RegOpnd * Instr::FindRegUseSrc(StackSym *sym, IR::Opnd* src)
         RegOpnd * baseOpnd = indirOpnd->GetBaseOpnd();
         if (baseOpnd != nullptr && baseOpnd->m_sym == sym)
         {
-            return baseOpnd;
+            return true;
         }
         else if (indirOpnd->GetIndexOpnd() && indirOpnd->GetIndexOpnd()->m_sym == sym)
         {
-            return indirOpnd->GetIndexOpnd();
+            return true;
         }
     }
     else if (src->IsListOpnd())
     {
         IR::ListOpnd* list = src->AsListOpnd();
         for (int i = 0; i < list->Count(); ++i)
         {
-            IR::RegOpnd* reg = FindRegUseSrc(sym, list->Item(i));
-            if (reg)
+            if (HasSymUseSrc(sym, list->Item(i)))
             {
-                return reg;
+                return true;
             }
         }
     }
-    return nullptr;
+    else if (src->IsSymOpnd())
+    {
+        SymOpnd* symOpnd = src->AsSymOpnd();
+        if (symOpnd->GetSym() == sym)
+        {
+            return true;
+        }
+        if (symOpnd->IsPropertySymOpnd())
+        {
+            PropertySymOpnd* propertySymOpnd = symOpnd->AsPropertySymOpnd();
+            if (propertySymOpnd->GetObjectSym() == sym)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
 }
 
 
-IR::RegOpnd * Instr::FindRegUseDst(StackSym *sym, IR::Opnd* dst)
+bool
+Instr::HasSymUseDst(StackSym *sym, IR::Opnd* dst)
 {
     if (!dst)
     {
-        return nullptr;
+        return false;
     }
     if (dst->IsIndirOpnd())
     {
         IR::IndirOpnd *indirOpnd = dst->AsIndirOpnd();
         RegOpnd * baseOpnd = indirOpnd->GetBaseOpnd();
         if (baseOpnd != nullptr && baseOpnd->m_sym == sym)
         {
-            return baseOpnd;
+            return true;
         }
         else if (indirOpnd->GetIndexOpnd() && indirOpnd->GetIndexOpnd()->m_sym == sym)
         {
-            return indirOpnd->GetIndexOpnd();
+            return true;
         }
     }
     else if (dst->IsListOpnd())
     {
         IR::ListOpnd* list = dst->AsListOpnd();
         for (int i = 0; i < list->Count(); ++i)
         {
-            IR::RegOpnd* reg = FindRegUseDst(sym, list->Item(i));
-            if (reg)
+            if (HasSymUseDst(sym, list->Item(i)))
             {
-                return reg;
+                return true;
             }
         }
     }
-    return nullptr;
+    else if (dst->IsSymOpnd() && dst->AsSymOpnd()->IsPropertySymOpnd())
+    {
+        PropertySymOpnd* propertySymOpnd = dst->AsSymOpnd()->AsPropertySymOpnd();
+        if (propertySymOpnd->GetObjectSym() == sym)
+        {
+            return true;
+        }
+    }
+    else if (dst->IsSymOpnd())
+    {
+        SymOpnd* symOpnd = dst->AsSymOpnd();
+        if (symOpnd->GetSym() == sym)
+        {
+            return true;
+        }
+        if (symOpnd->IsPropertySymOpnd())
```

Comment:
You are checking `IsPropertySymOpnd()` twice now.
You probably should remove the `else if (dst->IsSymOpnd() && dst->AsSymOpnd()->IsPropertySymOpnd())` block above

Reply:
oh you're right, i didn't notice that block. i'll remove it

---
In reply to: [155049239](https://github.com/Microsoft/ChakraCore/pull/4330#discussion_r155049239) [](ancestors = 155049239)